pkg/jobconfig/files.go

```diff
@@ -443,7 +447,9 @@ func mergePostsubmits(old, new *prowconfig.Postsubmit) prowconfig.Postsubmit {
 func mergePeriodics(old, new *prowconfig.Periodic) prowconfig.Periodic {
 	merged := *new
 
-	merged.MaxConcurrency = old.MaxConcurrency
+	if old.MaxConcurrency != 0 {
```

Comment:
Technically this means that if someone tries to set their job to have no limit on concurrency (which is what `0` means) we will stop them. I think I'd rather check the labels to see if it's a publish job?

Reply:
Done.